cmd/minishift/cmd/hostfolder/add.go

```diff
@@ -20,45 +20,223 @@ import (
 	"fmt"
 	"runtime"
 
-	hostfolderActions "github.com/minishift/minishift/pkg/minishift/hostfolder"
+	"errors"
+	miniConfig "github.com/minishift/minishift/pkg/minishift/config"
+	hf "github.com/minishift/minishift/pkg/minishift/hostfolder"
+	"github.com/minishift/minishift/pkg/minishift/hostfolder/config"
+	"github.com/minishift/minishift/pkg/util"
 	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/minishift/minishift/pkg/util/strings"
 	"github.com/spf13/cobra"
 )
 
+const (
+	usage                = "Usage: minishift hostfolder add --type TYPE --source SOURCE --target TARGET HOST_FOLDER_NAME"
+	noSource             = "you need to specify the source of the host folder"
+	noTarget             = "you need to specify the target of the host folder"
+	noUserName           = "you need to specify a username"
+	noPassword           = "you need to specify a password"
+	noDomain             = "you need to specify the Windows domain"
+	unknownType          = "'%s' is an unknown host folder type"
+	nonSupportedTtyError = "not a tty supported terminal"
+	shareTypeFlag        = "type"
+	sourceFlag           = "source"
+	targetFlag           = "target"
+	optionsFlag          = "options"
+	interactiveFlag      = "interactive"
+	instanceOnlyFlag     = "instance-only"
+	usersShareFlag       = "users-share"
+)
+
 var (
 	instanceOnly bool
 	usersShare   bool
+	interactive  bool
+	shareType    string
+	source       string
```

Comment:
Source and target are globally derined, but used in hthe add code... why this choice?

Reply:
Ahh, now i see... They come from the arguments on start

Reply:
Not quite sure what you mean